src/ssl.c

```diff
@@ -20863,6 +20867,18 @@ unsigned long wolfSSL_X509_NAME_hash(WOLFSSL_X509_NAME* name)
         WOLFSSL_MSG("nothing to hash in WOLFSSL_X509_NAME");
         return 0;
     }
+  
+    size = wolfSSL_i2d_X509_NAME_canon(name, (byte**)&canon_name);
+    
+    if (size <= 0){
+        WOLFSSL_MSG("wolfSSL_i2d_X509_NAME_canon error");
+        return 0;
+    } 
+
+    XFREE(name->name, NULL, DYNAMIC_TYPE_STRING); 
+    name->name = canon_name;
```

Comment:
Are you sure that we want to substitute the buffer in `name->name`? I would be worried that this may break things elsewhere. IMO this should be changed where this buffer is created (looks like `CopyDecodedName` is one of those places). This could create weird dependencies where other functions don't know what to expect in `name->name`. @dgarske thoughts?

Reply:
I agree... re-use `name` if possible... You'll have to pass in the available buffer size. Don't free and allocate again.

Reply:
Ahh okay I can rework that way but how about the way I just changed it to? `wc_ShaHash(canon_name, size, digest)` ? So that the name buffer is untouched but hashes still match.

Reply:
This is perfect now. `name` is effectively constant. The only thing missing is the `free` for `canon_name`. Memory allocated by `wolfSSL_i2d_X509_NAME_canon` is being leaked.

Reply:
Correct `canon_name` needs free'd. Then this PR is good.

Reply:
Free'd.